src/rebar_app_utils.erl

```diff
@@ -287,16 +287,19 @@ expand_deps_sources(Dep, State) ->
     rebar_app_info:t() when
       Source :: rebar_resource_v2:source().
 update_source(AppInfo, {pkg, PkgName, PkgVsn, Hash}, State) ->
-    case rebar_packages:resolve_version(PkgName, PkgVsn, Hash,
+    update_source(AppInfo, {pkg, PkgName, PkgVsn, undefined, Hash}, State);
```

Comment:
This was needed to handle the case where we're running without locks, there may be a better place to tackle this but I didn't see one.